pkg/query-service/app/querier/querier.go

```diff
@@ -514,6 +515,13 @@ func (q *querier) QueryRange(ctx context.Context, params *v3.QueryRangeParamsV3,
 			} else {
 				results, err, errQueriesByName = q.runBuilderQueries(ctx, params, keys)
 			}
+			// in builder query, the only errors we expose are the ones that exceed the resource limits
+			// everything else is internal error as they are not actionable by the user
+			for name, err := range errQueriesByName {
+				if !chErrors.IsResourceLimitError(err) {
+					delete(errQueriesByName, name)
```

Comment:
Should we consider sending the error regardless ?

Reply:
How does it help user?

Reply:
it doesn't help the user, but the errors present in the response might help to get an quicker idea about the issue.

Reply:
Do you have an example of such errors that help to get a quicker idea about the issue from an end-user POV and make them actionable? I have not come across any such cases. Any build query we prepare must run, if it fails because of the resource the personalized error will be sent. The rest of the errors are ClickHouse errors that are not actionable by the user.

Reply:
Don't have an example from the end user pov unless they are using the api and want to debug the error directly from the response rather then checking the logs.

Reply:
We can go ahead with this.